src/Datagrid/Datagrid.php

```diff
@@ -221,7 +221,10 @@ public function getFilters()
 
     public function reorderFilters(array $keys)
     {
-        $this->filters = array_merge(array_flip($keys), $this->filters);
+        /** @var array<string, FilterInterface> $reorderedFilters */
+        $reorderedFilters = array_merge(array_flip($keys), $this->filters);
```

Comment:
Shouldn't be needed ; phpstan level 8 is running fine on master without this.

It might be better to let the datagrid class untouched in order to avoid conflict during the merge.

Reply:
The issue with this assignment was not triggered by PHPStan, but by Psalm.

Reply:
Indeed, there is 
```
<file src="src/Datagrid/Datagrid.php">
        <InvalidPropertyAssignmentValue occurrences="1">
            <code>array_merge(array_flip($keys), $this-&gt;filters)</code>
        </InvalidPropertyAssignmentValue>
    </file>
```
in the baseline of master.

Nice, we'll be able to remove it.

Reply:
@phansys, I think you should remove the phpdoc you added because the psalm issue is real.

```
$keys = ['bar', 'baz'];         
$elements = ['foo' => 'a', 'bar' => 'b'];
var_dump(array_merge(array_flip($keys), $elements));
```
will return
```
['bar' => 'b', 'baz' => 1, 'foo' => 'a']
```

The implementation should be modified, and we should decide what to do
- if a non existing key is passed (currently it's added to $elements and fail later so I would throw an early exception).
- if all existing keys are not passed (currently the others keys are added last).

Reply:
```
array_filter(array_merge(array_fill_keys($keys, null), $this->elements))
```
could fix the issue.

But it might better to decide to throw an exception if an extra keys was added or forgotten

Reply:
I've made https://github.com/sonata-project/SonataAdminBundle/pull/7287